### DISCUSSION TITLE
Koriki for RG35XX Beta 0.9.1

### DISCUSSION THREAD

**Discussion by acmeplus** [rg35xx-cfw/Koriki#28] [Announcements]
# Introduction
This is probably the last major public beta release of Koriki for the RG35XX before 1.0 is released. No major changes (other than bug fixes) are expected from this 0.9 to 1.0.  

Note that this is a beta release, so while the system is usable there may be issues that still need to be solved. Use the [Discussions Page](https://github.com/rg35xx-cfw/Koriki/discussions) to discuss any issues you may find, and create issues in the [Issues Page](https://github.com/rg35xx-cfw/Koriki/issues)

# Installation:
* Flash the koriki-rg35xx image to a SDCARD using balena etcher or a similar software
* Boot with the SDCARD. The first time the firmware will expand the second partition and copy some firmware files to that partition
* Important: this version now uses FAT32 for the second partition, so it should be accessible on Mac/PC without any dedicated software
* If you use a second SDCARD, the system will install those files in the last partition of the second card
* Note that the first time the system boots it will take a bit longer to expand the card and copy files over. Even when the menu is up, there will be a few seconds when the system will be a bit slow because it's adding the virtual memory image
* After the first boot the system will create some configuration folders as well as the roms and bios folders. Add your roms to the roms folder, and your bios to the roms folder.

# Update from previous installations:

* NOTE: If you installed v0.7 or higher, it's safe to use the update file. If you installed v0.6 or a previous version, you should install the full image instead since the partition tables have changed.
* Uncompress the file boot.tar.xz, and replace the contents of the first partition of your SDCARD image with this one (you should see files like uImage, ramdisk.img, boot folder, etc.)
* If you run into issues with content not starting or any other issue, rename your system folder (from the last partition of your main SDCARD or your second SDCARD), and start again.

# IMPORTANT
* If you use a second SDCARD that you have previously used with koriki for RG35XX, please rename your system folder, and start again.
* Some RG35XX boards have issues booting koriki. Download the archive ``bootfix_for_some_revisions.zip``, extract it, and copy the files ``uImage`` and ``kernel.dtb`` to the first Koriki SDCARD partition (overwriting the existing ones). Note that while that fix may solve boot issues, it may introduce some inconsistencies regarding controls like volume keys. 

# Charger behavior:
* The charging logic is slightly different from the one in the stock firmware (or Garlic). The system reimplements the charger logic that is performed via software and not via hardware in this system.
* Shutdown:
  * If the charger power cable is connected to the USB C port, once the system is shutdown it will boot into the charger mode. You will briefly see a battery symbol and the system will seem to be off but the charger led will be on (orange). To boot from this state, briefly press power key.
  * If the charger cable is not connected, the system will be off completely. To boot again from this state you need to press the power button a few seconds and the system will boot.
* Reboot:
  * The system will reboot as expected
 
# ChangeLog:
2023/07/07:
* Added Watara Supervision (potator) core
* Added Intellivision (freeing) core
* Added Atari 7800 (prosystem) core
* Added Wonderswan (mednafen_wsan) core
* Added Atari Lynx (handy, and mednafen_lynx) cores
* Added SGB (mesens) core (performance is still not good, use mgba instead)
* Fixed mupen64 config
* Fixed drastic config
* Fixed PPSSPP standalone config
* Fixed simplemenu roms paths for multiple systems: atari5200, atari7800, atarilynx, neogeocd, ngpc, SG-1000, wonderswan, etc
* Added simplemenu compatibility with batocera, Miyoo, and Garlic roms paths. Inserting a SD Card containing roms from any of those firmwares should now be detected correctly. System now detects thumbnails on both media/images (batocera) and Imgs (Garlic/Miyoo) folders.
* Fixed a bug that showed the mouse cursor when launching a rom
* Fixed a bug where simplemenu didn’t return to the same gamelist location after closing a rom/port

2023/07/04:
* Fixed controls getting stuck in one direction. This was mostly visible on shoot’em up games like 1944, Last Resort, etc.
* Removed evmapy dependency. Added joystick support to simplemenu so evmapy is no longer needed to create key events from joystick events
* Retroarch no longer has duplicated or incorrect events due to the removal of evmapy and new input mappings. 
* Retroarch key mappings/hotkeys have changed to be consistent with Koriki on the Miyoo Mini:
    * MENU + A: Pause
    * MENU + B: FastForward Toggle
    * MENU + X: Menu
    * MENU + Y: Reset Content
    * MENU + L1: Save Save State
    * MENU + R1: Load Save State
    * MENU + L2: Next Save State
    * MENU + R2: Previous Save State
    * START + SELECT: Quit Game
* Created SDL_GAMECONTROLLERCONFIG configuration file in ~/.config/gamecontrolerconfigdb.txt
* Fixed the simplemenu framebuffer glitch when launching roms/ports
* Fixed simplemenu background changing when selecting cores in simplemenu (pressing select to access game menu)
* Fixed simplemenu  favorites not being created
* Fixed first SDCard SHARE partition not getting properly expanded (not visible on MacOS, etc.)
* Added RNDIS support (thanks @JohnnyOnFlame). You need to use the adb_uImage and then create a script that calls usbmond.sh RNDIS.
* Initial wifi work (not yet working via configuration files). 
* Bumped version to Beta 0.9

2023/06/19
* Added timeout for ``POWER`` button press. System no longer shutdowns immediately when the button is pressed, now it needs to be pressed about 3s
* Fixed volume increasing to 100% when unit was shut down via power button
* Fixed brightness control combo (``MENU``+``VOL+/-``)
* Fixed audio volume keys if you use adb_uImage (mixer was different between adb_uImage and uImage)
* Fixed HDMI audio. Note that HDMI cable needs to be plugged in before boot for HDMI to work
* Added PCSX standalone configuration
* Added Devolutionx (Diablo) gamepad support
* Re-Added KM MAME 2003 xtreme core (removed by mistake in 0.7.3)
* Added libretro CROCODS core (Amstrad CPC)
* Added libretro quickness core (NES)
* ~~Added libretro bk (elektronika PDP)~~ (currently broken)
* Added libretro gearcoleco (ColecoVision)
* Added libretro GME (blargg's Game_Music_Emu)
* Added libretro RACE (NGPC)
* Added launcher for eduke32 (DukeNukem3D)
* Bumped version to Beta 0.8

2023/06/16
* Modified the FP flags from VFP3-D16 to NEON due to some performance issues detected in PCSX rearmed core. This should bring some performance improvements
* Recompiled all cores with the new architecture
* Updated full buildroot packages to synchronize to the latest batocera build
* Removed some unnecessary patches. Note that there may be some inconsistencies with previous releases, please report those in the discussions page, discord, or create issues if the problem is well defined
* Improved charging logic. When connecting a power cable when the console is off, the console will enter into charging mode without booting into koriki
* Added PCSX standalone (note, currently is not possible to exit the emulator, if anyone knows how to quit, please let me know)
* Added default mupen64 controls (you can exit the emulator with ``START``+``SELECT``)
* Added ATC theme in Spanish (Thanks to @albgr1979) (theme switching should now work)
* Enabled USB for controllers and some other devices. By default this release works with USB devices like gamepads, USB Wifi and USB Ethernet. However, please not that USB wireless is not currently configured.
  * Note: If you want to use ADB, copy over ``adb_uImage`` into  ``uImage`` 
* Improved input handling by updating the input driver to the latest garlic gpio module (quick movement from L to R or U to D should not leave the key stuck in the initial direction)
  * Note: there's a regression with the power button due to the new input driver. Use the simplemenu menu to shutdown/reboot

2023/06/13
* Added power button shutdown. Short press power button will shutdown the console. If the charger is connected it will boot into charger mode. From there another short press boots the console
* Fixed shutdown while charging
* Added brightness control shortcut with ``MENU``+``VOLUME+/VOLUME-``
* Fixed Sega32X launcher
* Changed AmstradCPC core to cap32

2023/06/12
* Update to fix an issue with the swap and share partitions being overwritten
* Reapplied the SDL patch that was not correctly applied (thanks again @johnnyonflame)
* Fixed typo in mastersystem path
* Added proper openbor launchers
* Added o2em launcher
* Fixed sega32x path

2023/06/11
* Added SNES mednafen-supafaust core (default for SNES). StarFox now runs close to 60fps without overclock or video threading
* Added MAME km_mame_2003_xtreme that runs much better for some roms (default for MAME)
* Fixed swap virtual memory generation, it has its own dedicated partition now, should be much more efficient and not impact FAT32 partitions
* Added additional ports location
* Added SDL VT patch to solve some issues with some ports not returning to simplemenu (thanks @johnnyonflame)

2023/06/08
* Fixed simplemenu NO GAMES FOUND message. Now it will go back to the last selection after a game is closed or after restart
* Changed data partition to FAT32 format
* Added controller configuration for standalone emulators (flycast, PPSSPP, mupen64plus, and Drastic)
* Changed default emulators, removed some missing emulators from the core list

2023/06/06
* First public release (alpha v0.5)
* Implemented global audio control (thanks to [adixal](https://codeberg.org/adixal) for his muOS joyaudio driver)
* Improved the charger logic. System now should properly respond to shutdown or reboot commands in koriki/simplemenu. Just go to settings -> Session, press left/right until you see either shutdown or reboot and press A to confirm. See charger behavior below for details.
* Updated flycast and ppsspp cores to the latest release
* Added drastic, flycast, and ppsspp application launchers. 

2023/06/04
* Updated mgba core
* Added cores: vba-m, gpsp, stella2014
* Added flycast standalone
* Fixed partition autoexpand

2023/06/03
* Updated launchers and menu sections so most systems should now work right out of the box
* Fixed charger led indicators
* Added od-commander
* Added missing flycast_libretro cores
* Added virtual memory back (should help to launch N64 titles)

2023/06/02
* Updated ``koriki-rg35xx_20230602.img.gz``
* Fixed battery charge indicator

2023/06/01
* Reworked system settings menu
* Added system overclock support
* Added cores back to the image
* Added battery charger logic (use reboot or shutdown in simplemenu to change behavior)
* System should work with either 1 sdcard or 2 sdcard (SD1 koriki, SD2 roms)
  * By default roms use the batocera roms names and assume those are located in ``/userdata/roms/<system>``
  * You can use garlic or OFW images for the roms in the second SDCARD slot, but you will need to modify the paths in ``/userdata/system/.simplemenu/sections_groups``
* SDCard partition auto expand is current broken

2023/05/31
* ~~Added koriki_share_rg35xx.zip archive. This archive is intended to be used for a second sdcard. Uncompress at the top of the archive.~~
* It contains the cores, as well as the folder structure to add roms to the system

2023/05/30
* First development release
* Image automatically generated
* SDCARD expansion
* External SDCARD compatibility
* Only a few set of cores included

## Working:
* Navigation, rom selection, launching roms
* Brightness control
* Battery indicator
* Initial volume control via keys (menu does not reflect control change)
* Volume control for all emulators (thank to [adixal](https://codeberg.org/adixal) for this)
* System overclock control
* Retroarch menu can be opened with the ``Menu`` key. Quit with ``START``+``SELECT``
* Content auto-copied when a new SDCARD is inserted
* Auto mounts last partition of a second sdcard if present
* USB ethernet
* External controllers (in retroarch)

## Not working:
* ~~By default the system will start showing a message saying "No Games Found". Just press B to go back and select your rom. This is a bug that needs to be fixed in this version of simplemenu~~
* Some menus, settings are still incomplete
* Some emulators need to be updated
* ~~Standalone emulators require key mappings to be updated~~
* ~~Input event issues with retroarch (map retroarch configuration manually)~~
* Application launchers (e.g. Wifi, need to be updated)
* ~~When the console starts or the menu is opened after a game is closed simplemenu shows there are no games available. Press B to go back and then you can go into the selection.~~
* ~~Controls for standalone emulators are not properly mapped~~
* ~~Networking is still not configured~~
* ~~External controllers are not working at this point (kernel is in adb mode by default)~~
* ~~Selecting a core in simplemenu changes the background color (mostly cosmetic)~~

## Tweaking configuration:
* HOME is ``/userdata/system`` (also alias to /mnt/SDCARD)
* Simplemenu files are in ``$HOME/.simplemenu``
* Sections are in ``$HOME/.simplemenu/section_groups`` (self explanatory)
* Emulator launchers are in ``$HOME/.simplemenu/launchers`` (self explanatory, but look for the ones like neogeo_libretro that have been already adapted) 
* By default the firmware expects batocera-like roms partitions (with folders under roms like neogeo, mame, megadrive, etc.). It's possible add additional roms paths in ``$HOME/.simplemenu/section_groups`` .ini files. For instance, for master system you can see an example of batocera path (/userdata/roms/mastersystem) and Miyoo Koriki path (/userdata/Roms/MS). 
```[MASTER SYSTEM]
execs = /userdata/system/.simplemenu/launchers/picodrive_libretro,/userdata/system/.simplemenu/launchers/genesis_plus_gx_libretro,/userdata/system/.simplemenu/launchers/gearsystem_libretr
romDirs = /userdata/roms/mastersystem/,/userdata/Roms/MS
romExts = .zip,.sms,.7z
```
* The snapshots/art/thumbnail previews are by default under ``media/images`` (e.g. neogeo/media/images/mslug.png). You can change that path in $HOME/.simplemenu/config.ini and change the value of ``media_folder = media/images`` (for instance ``media_folder = Img`` is the default Koriki/miyoo path).

<hr /><em>This discussion was created from the release <a href='https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v0.9.1'>Koriki for RG35XX Beta 0.9.1</a>.</em>

**Comment by penutpls** [comment]
Hey there!
I have just flashed Koriki onto a SDcard and put it into the RG35XX. It says that the first boot might take longer, but do you know how long? It has been going for a while but it's still on the boot image.
Also, was I supposed to allocate/format the second partition for ROMs myself or does it go automatically with the first boot. 
Thanks in advance!

**Reply by fender357** [reply]
Several things. You may need the modified files noted in the main document for the boot fix as some hardware has issues booting normally. 
Second, I found that I had to double check that my RG35xx was plugged into power as it wasn't charged up enough and would only show that initial loading logo and nothing else. 
For the second SD card, when the system boots the first time it will format that card and add all the directories. Then you can add your files to them.

**Reply by penutpls** [reply]
Thanks for the reply!
So, I'm only using one SDcard in TF1 of the RG35XX. Do I still not need to format the other part of the SDcard and Koriki will do it automatically, or should I allocate/format myself?
Also, my device was very low battery when trying this, so that might have been it.
Will try again with fully charged battery!

**Reply by fender357** [reply]
I'm not totally sure but I'm fairly certain it will add the directories onto the single SD card if you're using only one.

**Reply by acmeplus** [reply]
Koriki will format and expand the card.

**Comment by NamVu69** [comment]
Please fix the bug where plugging in a headphone jack and turn volume up/down then the device turns on speaker (headphone jack still plugs in)

**Reply by acmeplus** [reply]
Was not aware of that, will investigate.

**Reply by dmauri82** [reply]
In my rare motherboard always the sound output is by speaker and at the same time by headphones.

**Reply by NamVu69** [reply]
maybe it is hardware's fault. I didn't have the chance to test the stock firmware, since i upgraded to Koriki immediately when received.

**Reply by acmeplus** [reply]
There are two main different models of the RG35XX. The V6C is different internally than the rest of the other revisions. For the most part it behaves exactly the same, but volume keys, menu, start, select keys, as well as headphone detection is different. So if the headphone does not behave properly it does not mean your motherboard is bad, just that you may have one of those revisions.

**Reply by dmauri82** [reply]
With other distributions (GarlicOS) the speaker and headphones works well. When you connect the headphones automatically change. Same when you disconnect them

**Reply by acmeplus** [reply]
Garlic has provisions for dealing with the different hardware revisions. Some of those revisions have a different configuration for the headphone detection.

**Comment by dmauri82** [comment]
Will the 1.0 version release fix that some motherboards revision hangs without updating the files with "bootfix_for_some_revisions.zip"?
The problem is that this fix does not allow you to decrease the volume level. Any workaround?

**Reply by acmeplus** [reply]
It's a bit complicated, I have very limited time these days and I no longer have one of those units. I'll try to put together some test tools to gather some information from those revisions and try to understand what works.

Are the rest of the keys properly mapped? Can you use menu, start, and select in retroarch and koriki without issues?

**Reply by dmauri82** [reply]
All the rest keys are properly mapped in kokiri menu, also running a game or inside retroarch! Is the very rare that only volumes mapping has been lost! Tks for answer!

**Reply by dmauri82** [reply]
I don't know if I can help with something doing some tests for you. Please no hesitate to contact me!

**Reply by dmauri82** [reply]
Hi acmeplus!
I installed your other batocera IMG (not kokiri): 
Batocera 37-dev for RG35XX Beta v0.8.1]
https://github.com/rg35xx-cfw/rg35xx-cfw.github.io/releases/tag/rg35xx_batocera_beta_v0.8.1

The images only boot if I apply the files "bootfix_for_some_revisions.zip". But the surprise is that the volume buttons work properly! The rest of the mapping as well!

**Reply by acmeplus** [reply]
The volume mapping in batocera is different, I need to upload a test program here to get the volume keys identified on devices like yours so I can update the program that manages the audio.

**Reply by acmeplus** [reply]
Here's an updated boot fix. It's still not complete (power leds are not working) but should fix the boot issues as well as the volume keys not responding:

[bootfix_for_revision_v6c.zip](https://github.com/rg35xx-cfw/Koriki/files/12073720/bootfix_for_revision_v6c.zip)

**Reply by nitou2504** [reply]
Hi, first thank you for your amazing work on Koriki, I love being able to play nds and psp on this little handheld. Already tried the new fix and the volume buttons does work, the power led also works (green one right?). However, the brightness control menu+ vol. keys also modify the volume. It is interesting considering that before, while the volume keys didnt change the volume, they were able to modify the brightness (pressing menu).

**Reply by ghost** [reply]
With this bootfix power LED works, charging LED does not. Volume seems to change along with brightness, but brightness doesn't change along with volume, so for the time being it's possible to set them up kinda-independently.

**Reply by acmeplus** [reply]
Yes, that bootfix is not finished yet, it was mostly for users with that configuration to be able to at least boot. Charging is broken as well as headphone detection and probably other couple of things. Hopefully I'll fix all those issues for the next release.

**Comment by Beinzia** [comment]
Hello,

Volume control dont work for me, cannot up or down volume.

**Reply by acmeplus** [reply]
See https://github.com/rg35xx-cfw/Koriki/discussions/28#discussioncomment-6402728 for additional information.

**Reply by acmeplus** [reply]
See https://github.com/rg35xx-cfw/Koriki/discussions/28#discussioncomment-6471651 for a fix.

**Comment by AngeeelD** [comment]
Is anyone having issues with a second SD card?

I'm not able to see the SDCARD2 mounted, I've been exploring the tree directory in runtime using retroarch and an app utility, but I have not found anything about it.

Also according to the release notes, the Garlic Directory paths structure are compatible with this version (.91), I tested with a card that contains rooms with this directory structure and nothing appears :(

BTW, I noticed that sometimes when I flash the card and perform the first boot, sometimes it creates the directories system, roms, bios Shared partition, but some other times it does not.

**Reply by acmeplus** [reply]
Check the garlic card and see if you have a system folder, it may have not been created correctly. Delete that folder and try again. It make take a while the first time since the system is copying the folders and additional configuration files needed for koriki.

**Reply by AngeeelD** [reply]
I had no system folder at my garlic card.

I tried also, perform first boot with the the second SDCARD totally cleaned, nothing happeng with it

do you suggest have this second card in an specific format type? I've tried with FAT32, and EXFAT, but nothing is wrote in there at the first boot.

I also tried, with diff types brands of SDCARDs, but not lucky, I was thinking that probably the problems comes with my anbernic hardware, it may not be compatible with the firmware and that could be the reason why it is not mounted properly the second SDCARD  🤷🏽‍♂️

Just to let you know, I can put games in the second partition created in the 1st SDCARD without problems and everything works properly 🤟🏼

**Reply by acmeplus** [reply]
It's weird, the system detects a second card list of partitions and uses the last partition available for its content. That means if it's a garlic card it's the fourth partition that contains the Roms folder. If it's a batocera card it's typically the second partition. In any case, if you insert an empty FAT32 formatted card it should copy the content the first time it finds that there's no system folder there. Note that depending on the speed of the second card it may take a bit longer to copy the content, but should be somewhere between 10s to 30s since there are not that many files copied.

You can also try to copy the system, roms, and bios folder from the first card second partition, and copy that into the second partition, then see if that works.

**Reply by AngeeelD** [reply]
mmmnh, it is interesting what you said, my second card (garlic one) does not contain partitions, I mean, it is just one partition with roms, bios, cfw, saves, screenshots folders, let me try another clean installation, and see if I can achieve a proper partition system.

**Reply by AngeeelD** [reply]
now, after some attempts, I got your point, and I was able to read content from another SDCARD that using GarlicOS

basically, if we are using GarlicOS, the Koriki going to support the GarlicOS roms paths only if you are putting the main Garlic SDCARD (the one that usually is on slot1 when you are using GarlicOS), into the SDCARD slot2.

I do normally have the Second SDCARD only with one partition just with content data (game, saves, bios), and the first one with the system, using this way, I can switch between systems (taking advantage of that RG35xx has 2 SDCARD slots), without affecting my games and saves, and have the capability to test diff systems quickly.

e.g:
- SDCARD1 (GarlicOS) in RG35XX sd card slot1: 
  - Partition 1: 10.5MB: No idea what is here
  - Partition 2: 537MB: Contains System Files
  - Partition 3: 53.5MB: No idea what is here
  - Partition 4: rest of SDCARD space: Contains BIOS, CFW, Roms, Saves, etc folders
    - I normally do not store data here.

- SDCARD2 (Content) in RG35XX sd card slot2: 
  - Partition 1: Contains BIOS, CFW, Roms, Saves, etc folders.

** Garlic and Stock OS support direct read content data from SDCARD2, having this capability, I can switch between these two systems.

said that, Is there a way to support the following setup?:
I mean, by any configuration file, expecting not to read strictly a second partition (for BatoceraOS folder structure) or a fourth partition (for GarlicOS folder structure) like:
 
- SDCARD3 (KorikiOS) in RG35XX sd card slot1: 
  - Partition 1: BATOCERA: System Files.
  - Partition 2: SHARED: Content Data.
    - **I have no plan to store content data here but could be usable of course.

- SDCARD2 (Content) in RG35XX sd card slot2: 
  - Partition 1: Contains BIOS, CFW, Roms, Saves, etc folders.

**Reply by acmeplus** [reply]
Yeah, I realized that the automatic configuration only works for the first GarlicOS card and not for those who use a second SDCARD for Garlic. I'll try to add some logic to support that configuration, although it may not be included in 1.0 because I don't want to introduce regressions for what is currently working.

**Reply by AngeeelD** [reply]
it is fine @acmeplus, since is not a blocker to enjoy Koriki on rg35xx.

I just want to say that you are doing an amazing job!

**Comment by ghost** [comment]
File name for this version says "batocera" ;)

**Reply by acmeplus** [reply]
Yes, it's part of the build system, no plans to change it anytime soon :) This koriki version is built on top of batocera.

**Reply by ghost** [reply]
That's perfectly fine ;)

**Comment by Beinzia** [comment]
When I want to start my RG35 this happened![image](https://github.com/rg35xx-cfw/Koriki/assets/117917661/ef377352-0f7f-45ed-a88c-26821f76c480)

**Reply by fender357** [reply]
I had this happen, too, and I just needed to try powering it on while it was plugged into power. The battery was just too low for it to start even though it seems like it should've been enough.

**Reply by AngeeelD** [reply]
I had the same problem with a low-speed SDCARD, I tried with a better one and works on the first try.

**Comment by blazinnerd1** [comment]
My Dpad won't work in games or retroarc, works fine in the menu tho. Fresh install. Thank you!

**Reply by acmeplus** [reply]
Are you using a second SDCARD that previously had been used for koriki or batocera? If that's the case, remove or move the system folder and try to boot again. The system will re-create the system folder during the first boot.

If you don't have a second SDCARD, then it's probably a different issue. Try to load retroarch directly from Apps & Games -> Apps -> Retroarch and test. Also try to start one of the test roms included, Old Towers and see if controls respond.

**Comment by ewerton680** [comment]
How do I make the ppsspp load an external save file?

**Comment by martijn189** [comment]
I believe the instructions for Retroarch key binding is wrong. Save = menu +R1, load = menu + L1. Really looking forward for version 1.0

**Reply by acmeplus** [reply]
Thanks, will double check before 1.0. The mappings tried to match the ones in Koriki for the Miyoo Mini, but I may have mapped them incorrectly.

**Comment by lesterbryan** [comment]
Tried Openbor games but controls do not work..
Where can I configure it?

**Reply by acmeplus** [reply]
With some of the optimizations for the controls in v0.9.1 evmapy was not longer started but it's actually needed for OpenBOR to run. You can modify the file ``system/.simplemenu/launchers/OpenBOR4432_sa``

with this content (basically add the two lines related to evmapy):
```bash
#!/bin/sh

export HOME=/userdata/system
cd $HOME

evmapy &
        
OpenBOR4432 "$1"
                
killall evmapy
```

**Comment by lesterbryan** [comment]
I have 2 sd cards that have diff games on them..
when the main sd card is inserted my games on it are detected but when both sd cards are inserted, games on main sd card is not detected anymore

**Reply by acmeplus** [reply]
That’s by design. When the system detects a second sdcard it uses only that one and the second partition of the main sd card is ignored.

**Comment by lesterbryan** [comment]
My battery went from 47% to 10% in minutes?
Not sure if it is a battery indicator issue...is there a fix I can do check on this?

**Reply by acmeplus** [reply]
Most likely it was not fully charged or you have a different battery capacity than the default set in the dtb. I don't recall if 0.9.1 comes with default 2100mAh or 2600mAh but you may want to experiment replacing ``kernel.dtb`` with ``2100mAh.dtb`` or ``2600mAh.dtb``, just make a copy of kernel.dtb first and copy and rename any of those as kernel.dtb.

**Reply by yulan233** [reply]
Is it possible to create other capacity configurations, or can it be set in one configuration? Because the battery has been changed to other capacities, such as 3500 or 2800.

**Comment by kramzzie06** [comment]
not showing the second partition. so now i cant copy the roms

**Reply by kramzzie06** [reply]
how to show shared partition

**Reply by lesterbryan** [reply]
watch some instructions on how to install GarlicOS...it is the same..its how I did mine..and its my first time doing this kind of stuff...they will also tell how to show the hidden partitions..

**Comment by kramzzie06** [comment]
No thumbnail for my roms

**Reply by lesterbryan** [reply]
Did you make a separate folder named "Imgs" inside the specific rom folders?if not make one...example inside GBA rom make a folder named "Imgs" then put your game humbnails their

**Reply by acmeplus** [reply]
Yeah, by default the system expects the thumbnails to be png files that to match the rom name but with .png extension. So if you have something like ``roms/gb/game.zip``, it looks for ``roms/gb/media/images/game.png`` or ``roms/gb/Imgs/game.png``.

**Comment by paladinperry** [comment]
Thanks for all the hard work you're throwing into this guys.
I love batocera and I'm eager to run this once my RG35XX arrives as the sole OS
Do you have an ETA on the v1.0 ?
Obviously I'm not expecting N64 or PSP to run but if I get a single saturn game to run I'll feel like a king haha
honestly thank you for working so hard on this, I'm so glad this project is here now that Bat-lite is dead

**Comment by netsba** [comment]
Thank you very much for the work done. I would just like to know if the project is still current or it is abandoned ? Merci

**Reply by acmeplus** [reply]
No, not abandoned at all, simply a mixture of vacation, real work, and other projects. 

Here's an update I posted a few days ago about what's eating up most of my time: 
https://ko-fi.com/post/Simplermenu-H2H0P654W

The goal for 1.0 is not to include simplermenu+ yet but that's the end goal for this koriki for rg35xx firmware. 

Specifically about koriki 1.0, the work is focused on fixing control issues with things like openbor, etc. and hopefully to update the cores to the latest versions. The main issue with 1.0 is to get the unified kernel to work with all the rg35xx revisions. There was a previous post where I explain that I had to purchase another v6.0C to figure out the issue with those boards:

https://ko-fi.com/post/Status-update-D1D0NYM5L

**Reply by netsba** [reply]
Thank you very much for the answer and I am very happy that the project continues. Take all your time to make us a good release, no one is in a hurry to you. Just from time to time reassures us by giving us news.

**Comment by acmeplus** [comment]
A new interim update is now out. This is only a system (all the configurations used for the system) update, so requires 0.9.1 before updating. 

More details in the [v0.9.2-test release page](https://github.com/rg35xx-cfw/Koriki/releases/tag/koriki_rg35xx_v0.9.2-test)
